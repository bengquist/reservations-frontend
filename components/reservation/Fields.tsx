import * as React from "react";
import DatePicker from "react-datepicker";
import Section from "../ui/Section";
import styled from "styled-components";

type Values = {
  name: string;
  hotelName: string;
  arrivalDate: Date;
  departureDate: Date;
  errorMessage: string;
};

type Props = {
  values: Values;
  onInput: (value: any) => void;
};

const Fields: React.FunctionComponent<Props> = ({ onInput, values }) => {
  const { name, hotelName, arrivalDate, departureDate } = values;

  return (
    <>
      <Section
        label="Reservation Name:"
        placeholder="Blake"
        value={name}
        setValue={(value: string) => onInput({ name: value })}
      />
      <Section
        label="Hotel:"
        placeholder="Hampton Inn & Suites"
        value={hotelName}
        setValue={(value: string) => onInput({ hotelName: value })}
      />
      <DateContainer>
        <Section label="Arrival Date:" value={Number(arrivalDate)}>
          <DatePicker
            selected={arrivalDate}
            onChange={(value: Date) => onInput({ arrivalDate: value })}
          />
        </Section>
        <Section label="Departure Date:" value={Number(departureDate)}>
          <DatePicker
            selected={departureDate}
            onChange={(value: Date) => onInput({ departureDate: value })}
          />
        </Section>
      </DateContainer>
    </>
  );
};

export default Fields;

const DateContainer = styled.div`
  display: grid;
  grid-template-columns: repeat(2, 1fr);
  grid-gap: 2rem;

  @media (max-width: 500px) {
    grid-template-columns: 1fr;
  }
`;
