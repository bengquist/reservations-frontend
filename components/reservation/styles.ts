import styled, { css } from "styled-components";

export const Form = styled.form`
  display: grid;
  grid-template-columns: repeat(2, 1fr);
  grid-gap: 2rem;

  @media (max-width: 500px) {
    grid-template-columns: auto;
  }
`;

export const ButtonContainer = styled.div`
  display: grid;
  grid-template-columns: 1fr 1fr;
  grid-gap: 0.5rem;
  justify-self: end;
  align-self: end;
`;

export const secondaryStyles = css`
  background: white;
  color: black;

  :focus {
    color: white;
  }
`;

export const Message = styled.label`
  background: ${props => props.theme.colors.accent};
  grid-column: 1/3;
  grid-row: 3;
  color: white;
  border-radius: 5px;
  padding: 5px 20px;
  margin: 0 auto;
`;
