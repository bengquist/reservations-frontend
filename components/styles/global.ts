import { createGlobalStyle } from "styled-components";

export const GlobalStyle = createGlobalStyle`
  html {
    font-size: 10px;
  }
  body {
    font-size: 1.6rem;
  }
  :root {
    --step-up-5: 2em;
    --step-up-4: 1.7511em;
    --step-up-3: 1.5157em;
    --step-up-2: 1.3195em;
    --step-up-1: 1.1487em;
    /* baseline: 1em */
    --step-down-1: 0.8706em;
    --step-down-2: 0.7579em;
    --step-down-3: 0.6599em;
    --step-down-4: 0.5745em;
    --step-down-5: 0.5em;
    /* Colors */
    --header: rgb(0,0,0);
  }
  @font-face {
    font-family: system;
    font-style: normal;
    font-weight: 300;
    src: local(".SFNSText-Light"), local(".HelveticaNeueDeskInterface-Light"), local(".LucidaGrandeUI"), local("Ubuntu Light"), local("Segoe UI Light"), local("Roboto-Light"), local("DroidSans"), local("Tahoma");
  }
  body, h1, h2, h3, h4, h5, h6, p, ol, ul, input[type=text], input[type=email], button {
    margin: 0;
    padding: 0;
    font-weight: normal;
  }
  body, h1, h2, h3, h4, h5, h6, p, ol, ul, input[type=text], input[type=email], button {
    font-family: "system"
  }
  *, *:before, *:after {
    box-sizing: inherit;
  }
  ol, ul {
    list-style: none;
  }
  img {
    max-width: 100%;
    height: auto;
  }
  /* Links */
  a {
    text-decoration: none;
    color: inherit;
  &.active {
      text-decoration: none;
    }
  }
`;
