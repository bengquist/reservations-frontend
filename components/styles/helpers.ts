import { css } from "styled-components";

export const hoverState = css`
  transition: 0.2s;
  :hover,
  :active {
    background-color: ${({ theme }) => theme.colors.primary};
  }
`;
