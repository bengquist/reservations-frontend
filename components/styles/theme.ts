export type AppTheme = {
  colors: {
    primary: string;
    secondary: string;
    accent: string;
    gray: string;
  };
};

export type AppThemeColor = keyof AppTheme["colors"];

export const theme: AppTheme = {
  colors: {
    primary: "#DAF2FC",
    secondary: "#009CDE",
    accent: "#104C97",
    gray: "#E8E8E8"
  }
};
