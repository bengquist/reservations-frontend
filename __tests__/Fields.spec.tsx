import * as React from "react";
import { shallow, mount } from "enzyme";
import Fields from "../components/reservation/Fields";
import { ThemeProvider } from "styled-components";
import { theme } from "../components/styles/theme";

const defaultValues = {
  name: "",
  hotelName: "",
  arrivalDate: new Date("06-16-2019"),
  departureDate: new Date("06-17-2019"),
  errorMessage: ""
};

const setup = () =>
  shallow(<Fields values={defaultValues} onInput={jest.fn} />);

describe("Fields", () => {
  it("renders and matches snapshot", () => {
    const wrapper = setup();

    expect(wrapper).toMatchSnapshot();
  });

  it("renders section inputs", () => {
    const wrapper = setup();

    const sections = wrapper.find("Section");

    expect(sections.length).toEqual(4);
  });
});
