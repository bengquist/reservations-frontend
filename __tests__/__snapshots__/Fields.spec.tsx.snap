// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Fields renders and matches snapshot 1`] = `
<Fragment>
  <Section
    label="Reservation Name:"
    placeholder="Blake"
    setValue={[Function]}
    value=""
  />
  <Section
    label="Hotel:"
    placeholder="Hampton Inn & Suites"
    setValue={[Function]}
    value=""
  />
  <styled.div>
    <Section
      label="Arrival Date:"
      value={1560661200000}
    >
      <DatePicker
        allowSameDay={false}
        dateFormat="MM/dd/yyyy"
        dateFormatCalendar="LLLL yyyy"
        disabled={false}
        disabledKeyboardNavigation={false}
        dropdownMode="scroll"
        inlineFocusSelectedMonth={false}
        monthsShown={1}
        nextMonthButtonLabel="Next month"
        onBlur={[Function]}
        onChange={[Function]}
        onClickOutside={[Function]}
        onFocus={[Function]}
        onInputClick={[Function]}
        onInputError={[Function]}
        onKeyDown={[Function]}
        onMonthChange={[Function]}
        onSelect={[Function]}
        onYearChange={[Function]}
        preventOpenOnFocus={false}
        previousMonthButtonLabel="Previous Month"
        readOnly={false}
        renderDayContents={[Function]}
        selected={2019-06-16T05:00:00.000Z}
        shouldCloseOnSelect={true}
        showMonthYearPicker={false}
        showTimeInput={false}
        showTimeSelect={false}
        strictParsing={false}
        timeCaption="Time"
        timeInputLabel="Time"
        timeIntervals={30}
        withPortal={false}
      />
    </Section>
    <Section
      label="Departure Date:"
      value={1560747600000}
    >
      <DatePicker
        allowSameDay={false}
        dateFormat="MM/dd/yyyy"
        dateFormatCalendar="LLLL yyyy"
        disabled={false}
        disabledKeyboardNavigation={false}
        dropdownMode="scroll"
        inlineFocusSelectedMonth={false}
        monthsShown={1}
        nextMonthButtonLabel="Next month"
        onBlur={[Function]}
        onChange={[Function]}
        onClickOutside={[Function]}
        onFocus={[Function]}
        onInputClick={[Function]}
        onInputError={[Function]}
        onKeyDown={[Function]}
        onMonthChange={[Function]}
        onSelect={[Function]}
        onYearChange={[Function]}
        preventOpenOnFocus={false}
        previousMonthButtonLabel="Previous Month"
        readOnly={false}
        renderDayContents={[Function]}
        selected={2019-06-17T05:00:00.000Z}
        shouldCloseOnSelect={true}
        showMonthYearPicker={false}
        showTimeInput={false}
        showTimeSelect={false}
        strictParsing={false}
        timeCaption="Time"
        timeInputLabel="Time"
        timeIntervals={30}
        withPortal={false}
      />
    </Section>
  </styled.div>
</Fragment>
`;
